std/mem.zig

```diff
@@ -515,11 +524,10 @@ pub fn writeInt(buf: []u8, value: var, endian: builtin.Endian) void {
 }
 
 pub fn writeIntBE(comptime T: type, buf: *[@sizeOf(T)]u8, value: T) void {
-    assert(T.bit_count % 8 == 0);
-    const uint = @IntType(false, T.bit_count);
-    if (uint == u0) {
-        return;
+    if (comptime T.bit_count % 8 != 0) {
+      return @compileError("Only integer types with bitwidth in multiples of 8 are allowed; passed " ++ @typeName(T));
```

Comment:
Why do we need this error? The write functions should work fine with arbitrary bit widths.

Reply:
I don't believe they work with arbitrary bit widths?
@thejoshwolfe had this assert:
`assert(T.bit_count % 8 == 0);`
which I made into a @compileError so that we can detect it at comptime instead of at runtime.